### PR TITLE
	[FEATURE] Tickets can be requested (created).

### PR DESCRIPTION
**Pull request by joelhawksley**
screenshot 2015-10-02 14 54 44

### CODE REVIEW

File: app/models/ticket.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Ticket < ActiveRecord::Base
   belongs_to :app
   has_many :acceptance_criteria
+  belongs_to :requester, class_name: "User", foreign_key: "requester_id"
```
Comment (DavidDufresne) [inline app/models/ticket.rb]: Do you need to specify the foreign key if it's just `<association_name>_id`?

File: app/models/user.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class User < ActiveRecord::Base
   has_and_belongs_to_many :apps
+  has_many :requested_tickets, foreign_key: "requester_id", class_name: "Ticket"
```
Comment (DavidDufresne) [inline app/models/user.rb]: So perhaps I should call my association `owned_tickets`.

File: app/models/ticket.rb (hunk 1 above)
Comment (joelhawksley) [inline app/models/ticket.rb]: You're right!

File: app/models/user.rb (hunk 2 above)
Comment (joelhawksley) [inline app/models/user.rb]: I think it's important to emphasize the difference.

File: app/controllers/tickets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class TicketsController < ApplicationController
+  before_action :ensure_user
+
+  def create
+    if current_user.requested_tickets.create(app: current_app, description: params[:description])
```
Comment (DavidDufresne) [inline app/controllers/tickets_controller.rb]: Could we maybe use strong params here?

**Comment by DavidDufresne** [comment]
:+1:

**Comment by aesnyder** [comment]
LGTM :+1:

**Pull request merged**